src/routes/auth.js

```
@@ -0,0 +1,222 @@
+// src/routes/auth.js
+const { Router } = require('express');
+const prisma = require('../services/prisma');
+const bcrypt = require('bcryptjs');
+const { randomBytes } = require('crypto');
+const {
+  sendActivationEmail,
+  sendPasswordResetEmail,
+} = require('../services/email');
+const {
+  issueAccessToken,
+  issueRefreshToken,
+  rotateRefreshToken,
+  pruneExpired,
+} = require('../services/tokenService');
+const guest = require('../middleware/guest');
+const auth = require('../middleware/auth');
+
+const router = Router();
+
+/**
+ * REGISTER (guest-only)
+ * - cria usuario com activationToken + activationTokenExpires
+ * - envia e-mail de ativação (NO_EMAIL=1 => loga no console)
+ */
+router.post('/register', guest, async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const activationToken = randomBytes(32).toString('hex');
+  const activationTokenExpires = new Date(Date.now() + 24 * 60 * 60 * 1000); // 24h
+  const passwordHash = await bcrypt.hash(password, 10);
+
+  await prisma.user.create({
+    data: { name, email, passwordHash, activationToken, activationTokenExpires },
+  });
```

You call `prisma.user.create` without first checking whether a user with the same email already exists. Because the DB has a unique index on email, creating a duplicate will raise a DB error rather than returning a controlled 409 response. Add an existence check (or catch the unique constraint error and translate it) so you can return a helpful 409 response.